app/main/views/suppliers.py

```diff
@@ -119,6 +119,40 @@ def view_signed_agreement(supplier_id, framework_slug):
     )
 
 
+@main.route('/suppliers/<supplier_id>/agreements/<framework_slug>/on-hold', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def put_signed_agreement_on_hold(supplier_id, framework_slug):
+    supplier_framework = data_api_client.get_supplier_framework_info(supplier_id, framework_slug)['frameworkInterest']
+    data_api_client.put_signed_agreement_on_hold(supplier_framework['agreementId'], current_user.email_address)
+
+    flash(supplier_framework['declaration']['nameOfOrganisation'], 'put on hold')
+
+    return redirect(url_for(
```

Comment:
Yeah, I'd been working with just returning the agreement that was just put onhold/approved with the view that finding the next one to be processed was another story. I'm gonna leave it just redirecting to the `admin/agreements/g-cloud-8` page